solr/core/src/java/org/apache/solr/cloud/api/collections/assign/MoveReplicaRequest.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.cloud.api.collections.assign;
+
+/**
+ *
+ */
+public interface MoveReplicaRequest extends AssignRequest {
```

Comment:
What is the plugin asked to decide/compute here?

Reply:
In some cases you know exactly where you want to move a replica. In other cases you're asking for the best place other than the current place. This supports both scenarios - if `targetNode` is specified then this translates directly to a MOVEREPLICA cmd without any computation. If `targetNode` is null then the plugin needs to compute the best new placement.